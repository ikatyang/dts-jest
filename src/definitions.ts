import * as ts from 'typescript';

export type TriggerDescription = null | string;

export interface ITrigger {
  line: number;
  kind: TriggerKind;
  description: TriggerDescription;
}

export interface ITriggerDescriptions {
  [line: number]: TriggerDescription;
}

export enum TriggerKind {
  None,
  Skip,
  Only,
  Show,
}

export interface ITSConfig {
  extends?: string;
  compilerOptions?: ts.CompilerOptions;
}

export type IRawTSConfig = undefined | string | ts.CompilerOptions;

export interface ISelfConfig {
  tsconfig?: IRawTSConfig;
  reporter?: string;
}

export const default_reporter_template = '\nInferred\n\n{{expression,2}}\n\nto be\n\n{{snapshot,2}}';

export interface IJestConfig {
  globals: {
    _dts_jest_: ISelfConfig;
  };
}

export const config_namespace = '_dts_jest_';
export const runtime_namespace = '_dts_jest_runtime_';

export const trigger_regex = /^\s*\/\/\s*@dts-jest\b(:\S*)?\s*(.+)?\s*$/;
export enum TriggerMatchIndex {
  Input,
  Flag,
  Description,
}
export type ITriggerMatchArray = [
  string,
  string | undefined,
  string | undefined
];

export interface IRuntimeSnapshotParameters {
  line: number;
}
export interface IRuntimeDescriptionParameters {
  line: number;
}
export interface IRuntimeReportParameters {
  line: number;
}

export interface IRuntimeMethods {
  report(parameters: IRuntimeReportParameters): string;
  snapshot(parameters: IRuntimeSnapshotParameters): string;
  description(parameters: IRuntimeDescriptionParameters): string;
}
