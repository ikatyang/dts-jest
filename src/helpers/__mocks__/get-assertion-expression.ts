export const get_assertion_expression = () => '[assertion-expression]';
