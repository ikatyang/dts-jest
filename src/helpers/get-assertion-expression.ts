import {
  runtime_namespace,
  IRuntimeReportParameters,
  IRuntimeSnapshotParameters,
  TriggerKind,
} from '../definitions';

export const get_assertion_expression = (kind: TriggerKind, line: number) => {
  if (kind === TriggerKind.Show) {
    const report_parameters: IRuntimeReportParameters = {line};
    const report_value = `${runtime_namespace}.report(${JSON.stringify(report_parameters)})`;
    return `console.log(${report_value})`;
  }
  const snapshot_parameters: IRuntimeSnapshotParameters = {line};
  const snapshot_value = `${runtime_namespace}.snapshot(${JSON.stringify(snapshot_parameters)})`;
  return `expect(${snapshot_value}).toMatchSnapshot()`;
};
