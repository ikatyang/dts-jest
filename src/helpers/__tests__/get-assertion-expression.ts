jest.unmock('../get-assertion-expression');

import {TriggerKind} from '../../definitions';
import {get_assertion_expression} from '../get-assertion-expression';

const line = 1;

it('should return console.log() while kind is show', () => {
  expect(get_assertion_expression(TriggerKind.Show, line)).toMatchSnapshot();
});

it('should return expect().toMatchSnapshot() while kind is not show', () => {
  expect(get_assertion_expression(TriggerKind.None, line)).toMatchSnapshot();
});
