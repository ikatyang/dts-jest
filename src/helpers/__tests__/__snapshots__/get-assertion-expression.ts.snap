// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return console.log() while kind is show 1`] = `"console.log(_dts_jest_runtime_.report({\\"line\\":1}))"`;

exports[`should return expect().toMatchSnapshot() while kind is not show 1`] = `"expect(_dts_jest_runtime_.snapshot({\\"line\\":1})).toMatchSnapshot()"`;
